modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -196,35 +219,68 @@ Query percolateQuery(PercolateQuery.QueryStore queryStore, BytesReference docume
             return new PercolateQuery(queryStore, documentSource, candidateMatchesQuery, searcher, verifiedMatchesQuery);
         }
 
-        Query createCandidateQuery(IndexReader indexReader) throws IOException {
+        Query createCandidateQuery(IndexReader indexReader, DocumentMapper docMapper) throws IOException {
             List<BytesRef> extractedTerms = new ArrayList<>();
+            Map<String, List<BytesRef>> encodedPointValuesByField = new HashMap<>();
+
             LeafReader reader = indexReader.leaves().get(0).reader();
             for (FieldInfo info : reader.getFieldInfos()) {
                 Terms terms = reader.terms(info.name);
-                if (terms == null) {
-                    continue;
+                if (terms != null) {
+                    BytesRef fieldBr = new BytesRef(info.name);
+                    TermsEnum tenum = terms.iterator();
+                    for (BytesRef term = tenum.next(); term != null; term = tenum.next()) {
+                        BytesRefBuilder builder = new BytesRefBuilder();
+                        builder.append(fieldBr);
+                        builder.append(FIELD_VALUE_SEPARATOR);
+                        builder.append(term);
+                        extractedTerms.add(builder.toBytesRef());
+                    }
                 }
-
-                BytesRef fieldBr = new BytesRef(info.name);
-                TermsEnum tenum = terms.iterator();
-                for (BytesRef term = tenum.next(); term != null; term = tenum.next()) {
-                    BytesRefBuilder builder = new BytesRefBuilder();
-                    builder.append(fieldBr);
-                    builder.append(FIELD_VALUE_SEPARATOR);
-                    builder.append(term);
-                    extractedTerms.add(builder.toBytesRef());
+                if (info.getPointDimensionCount() == 1) { // not != 0 because range fields are not supported
+                    PointValues values = reader.getPointValues(info.name);
+                    List<BytesRef> encodedPointValues = new ArrayList<>();
+                    values.intersect(new PointValues.IntersectVisitor() {
+                        @Override
+                        public void visit(int docID) throws IOException {
+                            throw new IllegalStateException("should not get here");
+                        }
+
+                        @Override
+                        public void visit(int docID, byte[] packedValue) throws IOException {
+                            encodedPointValues.add(new BytesRef(packedValue));
```

Comment:
actually I can simplify this code to be like this:

```
PointValues values = reader.getPointValues(info.name);
List<byte[]> encodedPointValues = new ArrayList<>();
encodedPointValues.add(values.getMinPackedValue().clone());
encodedPointValues.add(values.getMaxPackedValue().clone());
encodedPointValuesByField.put(info.name, encodedPointValues);
```

because only the lowest and highest values are used to build to range query.